### PR TITLE
Add Jazzy job configurations.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
Adds CI, documentation, release source, and binary jobs for ROS 2 Jazzy. Based on the ROS 2 Rolling configurations with: `sed -i 's/rolling/jazzy/g'`.

### CODE REVIEW

File: index.yaml [hunk 1]
```diff
@@ -61,6 +61,35 @@ distributions:
...
     source_builds:
       default: iron/source-build.yaml
+  jazzy:
+    ci_builds:
+      benchmark: jazzy/ci-benchmark.yaml
+      nightly-connext: jazzy/ci-nightly-connext.yaml
+      nightly-cross-vendor-connext-cyclonedds: jazzy/ci-nightly-cross-vendor-connext-cyclonedds.yaml
+      nightly-cross-vendor-connext-fastrtps: jazzy/ci-nightly-cross-vendor-connext-fastrtps.yaml
+      nightly-cross-vendor-connext-fastrtps-dynamic: jazzy/ci-nightly-cross-vendor-connext-fastrtps-dynamic.yaml
+      nightly-cross-vendor-cyclonedds-fastrtps: rollingjazzyci-nightly-cross-vendor-cyclonedds-fastrtps.yaml
```
Comment (clalancette) [inline index.yaml]:
```suggestion
      nightly-cross-vendor-cyclonedds-fastrtps: jazzy/ci-nightly-cross-vendor-cyclonedds-fastrtps.yaml
```

**Review by clalancette — CHANGES_REQUESTED** [review]

File: index.yaml (hunk 1 above)
Comment (nuclearsandwich) [reply index.yaml]: b9692e9c6fb8020ee2556cf490f6b009ff335f26

**Review by clalancette — APPROVED** [review]
Looks good to me, thanks for doing this!

**Review by marcoag — APPROVED** [review]

**Comment by nuclearsandwich** [comment]
Commands for deploying jazzy to the buildfarm:

```
# Dry-run to make sure things is good
❯ generate_all_jobs.py https://raw.githubusercontent.com/ros2/ros_buildfarm_config/ros2/index.yaml --ros-distro-names jazzy
# Review the output ...
❯ generate_all_jobs.py https://raw.githubusercontent.com/ros2/ros_buildfarm_config/ros2/index.yaml --ros-distro-names jazzy --commit
...
❯ generate_release_trigger_upload_jobs.py https://raw.githubusercontent.com/ros2/ros_buildfarm_config/ros2/index.yaml
```

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/jazzy-pre-release-is-ready-for-the-tutorial-party/37472/1

**Pull request merged**